localstack/aws/protocol/service_router.py

```diff
@@ -300,17 +300,19 @@ def determine_aws_service_name(
 
     # 5. check the query / form-data
     values = request.values
-    if "Action" in values and "Version" in values:
+    if "Action" in values:
         # query / ec2 protocol requests always have an action and a version (the action is more significant)
-        query_candidates = services.by_operation(values["Action"])
+        protocols = ["query", "ec2"]
+        query_candidates = services.by_protocol_and_operation(protocols, values["Action"])
```

Comment:
i think it would be better to leave the signature of the `by_operation` method unchanged. in principle i want to be able to look up operation names without _having_ to know the protocol in advance.

i think it would be better to get all candidates here, and then filter the candidates by their protocol if you only want services with the `query` or `ec2` protocol. it's a bit of extra work but keeps the service catalog API clean and simple.

Reply:
Okay I see. We discussed quickly this change with Alex, because by doing this, we had a rest-json service come up in the candidates list (`codestar-notifications`). We get only the names of the services as a result of `by_operation`, I'm not sure how we could filter out afterwards. Any idea? We could return the full object instead for the candidates, with all data about the service and not only the name. 
Or then add an optional `protocols` filter in `by_operation`, that is not mandatory. I'm open to all suggestions.

Reply:
just do something like
```python
candidates = [service for service in services.by_operation(values["Action"]) if services.get(service).protocol in ("ec2", "query")]
```
the performance overhead of this will be minimal and i think this simplifies things.

Reply:
Yes thanks a lot! Will change those asap! It does look much better.